2.6_testers2.ipynb

```diff
@@ -247,10 +299,19 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "In this section we will look at running sections of a unit test in parallel.\n",
-    "Look at the use of `fork` in the example below.\n",
-    "The chain `fork` methods allows these blocks of code to be executed at the same time, independent of each other.\n",
-    "The `join` method tells the test to wait until both `fork` blocks have completed before moving on to whatever comes next.\n"
+    "In this section we will look at running sections of a unit test concurrently. In order to do this we will introduce 4 new features of testers2.\n",
+    "\n",
+    "| method | description |\n",
+    "| :---   | :---        |\n",
+    "| fork   | launches a concurrent code block, additional forks can be run concurrently to this one via the .fork appended to end of the code block of the preceeding fork |\n",
+    "| join | re-unites multiple related forks back into a single thread |\n",
+    "| enqueueSeq | Continues to add (enqueue) elements from the `Seq` to a `Decoupled` input interface, one at a time, until the sequence is exhausted |\n",
+    "| expectDequeueSeq | Removes (dequeues) elements from a `Decoupled` output interface, one at a time, and compares each one to the next element of the `Seq` |\n",
+    "---\n",
+    "\n",
+    "In the example below two `fork`s are chained together, and then `join`ed. In the first `fork` block the `enqueueSeq` will continue to add elements until exhausted. The second `fork` block will `expectDequeueSeq` on each cycle when data is available.\n",
+    "\n",
+    ">The threads created by fork are run in a deterministic order, largely according to their order as specified in code, and certain bug-prone operations that depend on other threads are forbidden with runtime checks. \n"
```

Comment:
May be worth noting that only operations which interact through the testers2 simulator API (and in the future, some testers2 communication API) are checked for inter-thread effects. One can use shared variables all they like and testers2 won't know a thing.

Reply:
I think I'll take a crack at explaining this in later work. My goal with this PR is to increase early exposure to testers2

Reply:
Sure, that's fine.